src/arch/x86/archx86.rs

```diff
@@ -614,77 +625,17 @@ pub fn release_byte(_instr: &mut Instructionx86){
     _instr.cursor=_instr.cursor-1;
 }
 
-pub fn get_int_signed<T: Num + BitOr<Output=T> + NumCast + Copy>(
-    _instr: &mut Instructionx86,
-     _byte: &mut T)-> bool
+pub fn get_int<T: Num + NumCast + Copy + 'static>(
+    _instr: &mut Instructionx86) -> Option<T>
+    where u64: AsPrimitive<T>
 {
-    
-    let length = ::std::mem::size_of::<T>();  
-    disasm_debug!("get_int_signed() {:?}", length);
-    for offset in 0..length {
-        let mut _byte0: u8 = 0;
-        let address_offset = (_instr.cinfo.offset-_instr.cinfo.address) as usize;
-        let offset_size: usize = ((_instr.cursor+offset as u64)-_instr.cinfo.offset) as usize;
-        if offset_size+address_offset >= _instr.cinfo.size{
-            return false;
-        }
-        _byte0 = _instr.cinfo.code[offset_size+address_offset];
-        let new_num = (_byte0 as u64) << (offset as u64 * 8);
-        disasm_debug!("\tget_int byte {:x}", _byte0);
-        disasm_debug!("\tget_int shifted byte {:x}", new_num);
-
-        match length{
-            1=>{
-                let mut cast_byte: i8 =  NumCast::from(*_byte).unwrap();
-                cast_byte |= new_num as i8;
-                *_byte = NumCast::from(cast_byte).unwrap();
-            },
-            2=>{
-                let mut cast_byte: i16 =  NumCast::from(*_byte).unwrap();
-                cast_byte |= new_num as i16;
-                *_byte = NumCast::from(cast_byte).unwrap();
-            },
-            4=>{
-                let mut cast_byte: i32 =  NumCast::from(*_byte).unwrap();
-                cast_byte |= new_num as i32;
-                *_byte = NumCast::from(cast_byte).unwrap();
-            },
-            _=>{
-                let mut cast_byte: i64 =  NumCast::from(*_byte).unwrap();
-                cast_byte |= new_num as i64;
-                *_byte = NumCast::from(cast_byte).unwrap();
-            },
-        }
-    }   
-    _instr.cursor=_instr.cursor+length as u64;
-    return true;
+    let result = ::arch::x86::analyzex86::read_int::<T>(_instr.cursor as usize, _instr.cinfo.address as usize, &_instr.cinfo.code);
```

Comment:
`::arch::x86::analyzex86` *probably* isn't a good place for something as generic as `read_int` to live, but I wasn't sure where else to put it at the moment